src/pages/index.js

```diff
@@ -0,0 +1,302 @@
+
+api
+  .getAppInfo()
+  .then(([initialCards, userInfo]) => {
+    initialCards.forEach((item) => {
+      const cardEl = getCardElement(item);
+      cardsList.append(cardEl);
+    });
+
+    nameOnPage.textContent = userInfo.about;
+    descriptionOnPage.textContent = userInfo.name;
+
+    const profileAvatar = document.querySelector(".profile__avatar");
+    profileAvatar.src = userInfo.avatar;
+  })
+  .catch(console.error);
+
+const profileEditButton = document.querySelector("#edit-modal");
+const profileModal = document.querySelector("#edit-profile-modal"); // edit modal div
+const newPostModal = document.querySelector("#new-post-modal"); // image modal div
+
+const profileName = document.querySelector("#name-input");
+const profileDescription = document.querySelector("#description-input");
+const profileForm = document.querySelector(".modal__form"); // edit form
+const nameOnPage = document.querySelector("#profile-name");
+const descriptionOnPage = document.querySelector("#profile-description");
+const profileExitButton = profileModal.querySelector("#exit-button"); // exit button for both modals
+const profileSubmitBtn = profileModal.querySelector(".modal__submit-btn");
+
+const cardSubmitButton = newPostModal.querySelector(".modal__submit-btn");
+const profileAddButton = document.querySelector(".profile__add-btn");
+const modalImageForm = newPostModal.querySelector(".modal__form"); // image Form
+const imageLink = newPostModal.querySelector("#link-input");
+const caption = newPostModal.querySelector("#caption-input");
+const imageSaveButton = newPostModal.querySelector("#save-image");
+const newModalCloseButton = newPostModal.querySelector(".modal__exit-btn");
+
+const previewModal = document.querySelector("#preview-modal");
+const previewImage = previewModal.querySelector(".modal__image");
+const previewCaption = previewModal.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close");
+
+// avatar modal
+const avatarModalBtn = document.querySelector(".profile__avatar-btn");
+const avatarModal = document.querySelector("#edit-avatar-modal");
+const avatarCloseBtn = document.querySelector("#avatar-exit-button");
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+const avatarForm = document.querySelector("#edit-avatar-form");
+const avatarImg = document.querySelector(".profile__avatar");
+const avatarInput = document.querySelector("#avatar-input");
+
+//delete card Modal
+const deleteCardModal = document.querySelector("#delete-card-modal");
+const deleteForm = document.querySelector("#delete-card-form");
+const deleteExitBtn = document.querySelector(".modal__delete-exit-btn");
+const deleteCancel = document.querySelector("#delete-card-cancel");
+const deleteSubmitBtn = document.querySelector("#delete-submit-button");
+
+let selectedCard;
+let selectedCardId;
+
+newModalCloseButton.addEventListener("click", () => {
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.